Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-filter/ContestFilter.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useMemo, useState } from 'react';
+
+import Heading, { HeadingType } from '../../guidelines/headings/Heading';
+import List, { Orientation } from '../../guidelines/lists/List';
+import { FilterType, IFilter } from '../../../common/contest-types';
+import ExpandButton from '../../guidelines/buttons/ExpandButton';
+import Button, { ButtonSize, ButtonType } from '../../guidelines/buttons/Button';
+import concatClassNames from '../../../utils/class-names';
+
+import styles from './ContestFilter.module.scss';
+
+
+interface IContestFilterTypeProps {
+    values: IFilter[];
+    type: FilterType;
+    orientation?: Orientation;
+    onFilterClick: (filterId: number) => void;
+}
+
+const ContestFilter = ({
+    values,
+    type,
+    orientation,
+    onFilterClick,
+}: IContestFilterTypeProps) => {
+    const [ expanded, setExpanded ] = useState(false);
+    const maxFiltersToDisplayCount = 3;
```

Comment:
This can be extracted outside of the component.

Reply:
After extracting the expand functionality outside of the component and into the ContestFilters component, I have the problem that both filters - status and strategy - are expanded after clicking only one of them. For that reason the logic for the expansion in the ContestFilter component is internalized so that it is triggered independently for each filter.